### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by salvino080-coder** [pytorch/xla#9738]
#!/usr/bin/env python3
"""
Run a short synthetic training session and save xm.metrics_report() to file for analysis.
"""
import argparse
import torch
import torch.nn as nn
import torch.optim as optim
import time

try:
    import torch_xla.core.xla_model as xm
except Exception:
    xm = None

def run(output):
    device = xm.xla_device() if xm is not None else torch.device("cpu")
    model = nn.Sequential(nn.Flatten(), nn.Linear(3*32*32, 128), nn.ReLU(), nn.Linear(128, 10)).to(device)
    opt = optim.SGD(model.parameters(), lr=0.01)
    loss_fn = nn.CrossEntropyLoss()
    # synthetic small run
    for _ in range(5):
        x = torch.randn(16, 3, 32, 32).to(device)
        y = torch.randint(0, 10, (16,)).to(device)
        opt.zero_grad()
        out = model(x)
        loss = loss_fn(out, y)
        loss.backward()
        if xm is not None:
            xm.optimizer_step(opt)
            xm.mark_step()
        else:
            opt.step()
    # collect metrics
    metrics = ""
    if xm is not None:
        metrics = xm.metrics_report()
    with open(output, "w") as f:
        f.write(metrics)
    print("Saved XLA metrics to", output)

if __name__ == "__main__":
    import argparse
    p = argparse.ArgumentParser()
    p.add_argument("--output", default="/tmp/xla_metrics.txt")
    args = p.parse_args()
    run(args.output)
